xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Userlister.xml

```diff
@@ -0,0 +1,488 @@
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </id>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>2</number>
+        <prettyName>Macro name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($confluenceGroups = $stringtool.split("$!wikimacro.parameters.groups", ","))
+#set ($xwikiGroups = [])
+#foreach ($untrimmedConfluenceGroup in $confluenceGroups)
+  #set($confluenceGroup = $untrimmedConfluenceGroup.trim())
+  #if ($confluenceGroup == "*")
+    #set ($xwikiGroups = [])
+    #break
+  #end
+  #if ($confluenceGroup.startsWith("site-admins") || $confluenceGroup.startsWith("confluence-admins") || $confluenceGroup == "administrators")
+    #set ($discard = $xwikiGroups.add("XWiki.XWikiAdminGroup"))
+  #elseif ($confluenceGroup.startsWith("site-users") || $confluenceGroup.startsWith("confluence-users") || $confluenceGroup == "users")
+    #set ($discard = $xwikiGroups.add("XWiki.XWikiAllGroup"))
+  #else
+    #set ($discard = $xwikiGroups.add("XWiki.$confluenceGroup"))
```

Comment:
Again, you shouldn't make any assumption regarding the reference syntax. You need to resolve ``$confluenceGroup`` using the ``user`` resolver and then serialize using the ``local`` serializer.